src/bflib_video.c

```diff
@@ -968,6 +968,19 @@ TbPixel LbPaletteFindColour(const unsigned char *pal, unsigned char r, unsigned
     }
     return *o;
 }
+
+/**
+ * Scale a given value according to the game resolution (units_per_pixel).
+ * (returned value is rounded up to the next integer if ends in .5 or more)
+ * default is based on 640x400 (units_per_pixel = 16)
+ * so if units_per_pixel = 16 the this function returns (base_value * 1)
+ *
+ * @param base_value The value (or size) for the game running at 640x400. base_value should always be divisible by 2, for compatibility with 320x200.
+ */
+long scale_value_for_resolution(long base_value)
+{
+    return ((((units_per_pixel * base_value) >> 3) + (((units_per_pixel * base_value) >> 3) & 1)) >> 1);
```

Comment:
It is hard to read.
Is it `( x * units_per_pixel / 8 + (x * units_per_pixel / 8)) / 2`
So it is `round(x * units_per_pixel /16)` actually?
Why 1/16 ? is it "fix point scaling resolution"?
Could you please type some comment in code about what.

Reply:
Yes, this function is essentially `round(x * units_per_pixel /16)`
it is 1/16 because `units_per_pixel = 16` at 640x400 or 640x480 resolution.
So: 
* in old "DK Gold High Res mode" code - a value of "x" is a fixed amount
* in new KFX code - sending value "x" to this function returns "x" scaled to the current resolution (units_per_pixel).

If this were HTML: "x" used to be measured in **px**, and it is now measured in **em** (where 1 em = 1px at 320x200)

I'll work on the comments in the code, to explain it better.

Reply:
Comments improved.

If this function is the same as `round(x * units_per_pixel /16)` then do you think all of the changes shown in d553eda are OK?

I replaced calculations that round (or not) in different ways, but I think this new way is "correct"/more accurate at weird resolutions.

i.e. I replaced all of these:
* `(x*units_per_pixel+x)/16`
* `x * units_per_pixel / 16`
* `(x * units_per_pixel + 8) / 16`
with:
 `round(x * units_per_pixel /16)`

Reply:
ok